sessionctx/variable/session.go

```diff
@@ -1245,7 +1246,25 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 		s.MetricSchemaRangeDuration = tidbOptInt64(val, DefTiDBMetricSchemaRangeDuration)
 	case CollationConnection, CollationDatabase, CollationServer:
 		if _, err := collate.GetCollationByName(val); err != nil {
```

Comment:
If this collation is invalid, MySQL returns an error instead of using the default collation of current charset.
```
mysql> set collation_connection='23';
ERROR 1273 (HY000): Unknown collation: '23'
```